bundles/org.eclipse.equinox.p2.publisher.eclipse/src/org/eclipse/equinox/p2/publisher/eclipse/statefactory/StateBuilder.java

```diff
@@ -0,0 +1,923 @@
+/*******************************************************************************
+ * Copyright (c) 2003, 2020 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Danail Nachev -  ProSyst - bug 218625
+ *******************************************************************************/
+package org.eclipse.equinox.p2.publisher.eclipse.statefactory;
+
+import java.lang.reflect.Constructor;
+import java.util.*;
+import org.eclipse.osgi.internal.framework.EquinoxContainer;
```

Comment:
Should not import this internal for the constant.  And you seem to have an `EquinoxContainer` class in this `statefactory` you can use.